src/main/java/com/microsoft/azure/functions/worker/binding/ExecutionContextDataSource.java

```diff
@@ -64,4 +97,78 @@ public MethodBindInfo getMethodBindInfo() {
     public Class<?> getContainingClass() {
         return containingClass;
     }
+
+    private static Map<String, Parameter> addParameters(MethodBindInfo methodBindInfo){
+        Map<String, Parameter> map = new HashMap<>();
+        for (ParamBindInfo paramBindInfo : methodBindInfo.getParams()) {
+            map.put(paramBindInfo.getName(), paramBindInfo.getParameter());
+        }
+        return map;
+    }
+
+    @Override
+    public Optional<String> getParameterName(String annotationType){
```

Comment:
to make it clearer what should be passed in
```suggestion
    public Optional<String> getParameterName(String annotationSimpleClassName){
```

Reply:
let's discuss the usage of `Optional` in tomorrow's API review